grayskull/strategy/cran.py

```diff
@@ -0,0 +1,345 @@
+                chunk[i] = None
+        else:
+            if accumulating_continuations:
+                assert had_continuation
+                chunk[continued_ix] = continued_line
+                accumulating_continuations = False
+                continued_line = None
+                continued_ix = None
+
+    if had_continuation:
+        # Remove the None(s).
+        chunk = [c for c in chunk if c]
+
+    chunk.append("")
+
+    return chunk
+
+
+def yaml_quote_string(string):
+    """
+    Quote a string for use in YAML.
+
+    We can't just use yaml.dump because it adds ellipses to the end of the
+    string, and it in general doesn't handle being placed inside an existing
+    document very well.
+
+    Note that this function is NOT general.
+    """
+    return (
+        yaml.dump(string, Dumper=SafeDumper)
+        .replace("\n...\n", "")
+        .replace("\n", "\n  ")
+        .rstrip("\n ")
+    )
+
+
+# Due to how the metadata is rendered there can be
+# significant areas of repeated newlines.
+# This collapses them and also strips any trailing spaces.
+def clear_whitespace(string):
+    lines = []
+    last_line = ""
+    for line in string.splitlines():
+        line = line.rstrip()
+        if not (line == "" and last_line == ""):
+            lines.append(line)
+        last_line = line
+    return "\n".join(lines)
+
+
+# Reads the description file contents and formats them by
+# running other functions on the content and returns the dictionary.
+def read_description_contents(fp):
+    bytes_ = fp.read()
+    text = bytes_.decode("utf-8", errors="replace")
+    text = clear_whitespace(text)
+    lines = remove_package_line_continuations(text.splitlines())
+    return dict_from_cran_lines(lines)
+
+
+# Extracting the DESCRIPTION file from the downloaded package.
+def get_archive_metadata(path, verbose=True):
+    if verbose:
+        print("Reading package metadata from %s" % path)
+    if basename(path) == "DESCRIPTION":
+        with open(path, "rb") as fp:
+            return read_description_contents(fp)
+    elif tarfile.is_tarfile(path):
+        with tarfile.open(path, "r") as tf:
+            for member in tf:
+                if re.match(r"^[^/]+/DESCRIPTION$", member.name):
+                    fp = tf.extractfile(member)
+                    return read_description_contents(fp)
+    elif path.endswith(".zip"):
+        with zipfile.ZipFile(path, "r") as zf:
+            for member in zf.infolist():
+                if re.match(r"^[^/]+/DESCRIPTION$", member.filename):
+                    fp = zf.open(member, "r")
+                    return read_description_contents(fp)
+    else:
+        sys.exit("Cannot extract a DESCRIPTION from file %s" % path)
+    sys.exit("%s does not seem to be a CRAN package (no DESCRIPTION) file" % path)
+
+
+def get_cran_archive_versions(cran_url, session, package, verbose=True):
+    if verbose:
+        print(f"Fetching archived versions for package {package} from {cran_url}")
```

Comment:
Can you use log, or can you use ``print_msg`` from ``grayskull.cli.stdout`` please?